### ISSUE TITLE
Allow managers to easily find translate interface for FPPs

### ISSUE THREAD

**Issue by emmajane** [humanitarianresponse/site#374]
Currently the permissions are set to allow managers to translate fieldable panel panes. However; the only access point in to the list of FPPs appears to be via the admin panel (/admin/structure/fieldable-panels-panes/fieldable_panels_pane/fields) -- which managers do not have access to. As many groups have panes with the exact same name, it is very important that we limit the listing to the current space to avoid updating the wrong one.

I have investigated:

- looking for a translate button from within the Panels UI. (It's available if you go directly to the FPP page. e.g., https://www.humanitarianresponse.info/en/admin/structure/fieldable-panels-panes/view/2002 )
- updating the view which lists all content for a space to also include FPPs (node/%/content). Couldn't find a relevant relationship to "add" FPPs to the list.
- creating a new view. I couldn't find a contextual filter to limit the view to the current OG or node. "Bundle: type" is available as a contextual filter. See the view: fieldable_pane_entities (might be provided by Panels?).

**Comment by damienmckenna** [comment]
I'd be willing to work on a patch for this, if we could work out a reasonable way of handling it.

**Comment by emmajane** [comment]
@damienmckenna you are brilliant and wonderful! I promise to track you down on IRC to chat through this.

**Comment by wizonesolutions** [comment]
@emmajane @damienmckenna I'm working on something like this for a client. The way I'm planning to go about it is more or less what I say in this DA question: http://drupal.stackexchange.com/questions/192765/how-do-i-allow-editors-without-access-to-the-admin-to-translate-fieldable-panels

(The second approach.)

I'm going to push to be able to release the module. I can probably join forces to some extent, though. Let me know what you both figure(d) out.

**Comment by damienmckenna** [comment]
@wizonesolutions, that's awesome! I'd be completely willing to add functionality to FPP itself, I just haven't had the time to look into it.

**Comment by wizonesolutions** [comment]
@damienmckenna Perhaps, then, the best approach is this:

1. When I have the module done, I'll get the OK internally and then open an issue and attach it
1. We can then talk it out and see what makes sense.

In my mind, a proper implementation would be a little fancier or have more configurability (my plan is literally that if you turn on the module, it'll start doing its thing for FPPs). In an ideal world, there'd be some arrangement that'd allow easily changing between translations, maybe in i18n_variable style (links added to the form). There are some challenges, though, since it's not really "right" unless the parent node also gets redirected to the proper language, and then we somehow auto-open the FPP config screen. Probably doable with creative use of `$_SESSION`. Anyway, that's details for the issue queue.

**Comment by emmajane** [comment]
@wizonesolutions This is fantastic! Thank you! We may end up being an interesting test case for "creative" use of $_SESSION as we're sitting behind Varnish. I know this had some issues for language settings before I started working on the project, but I'm not sure how it was resolved.

**Comment by wizonesolutions** [comment]
Yeah, and that part would be more of a bonus anyway. Could alternatively
just use a query string. I actually didn't get much time on the project
this week, but I'll see how things pan out. If not this week, then next.
